gbpservice/nfp/orchestrator/config_drivers/heat_driver.py

```diff
@@ -445,6 +484,69 @@ def _get_all_heat_resource_keys(self, template_resource_dict,
                 resource_keys.append(key)
         return resource_keys
 
+    def _create_firewall_template(self, auth_token, service_details, stack_template):
+        provider = service_details['provider_ptg']
+
+        consuming_ptgs_details = service_details['consuming_ptgs_details']
+        consumer_eps = service_details['consuming_external_policies']
+
+        if (not consuming_ptg_details) and (not consumer_eps):
```

Comment:
Variable not define error, consuming_ptg_details
above define variable is consuming_ptgs_details